src/parser/SGTypes.ts

```diff
@@ -161,6 +161,21 @@ export class SGScript extends SGTag {
             return super.transpileBody(state);
         }
     }
+
+    public clone() {
+        return new SGScript(
+            { ...this.tag },
+            this.attributes.map(a => {
```

Comment:
You sure? It means the original tag/attributes sourcemap will be lost.
Aside from that it's cool.

Reply:
How would it lose the sourcemap? I'm cloning the range of each attribute as well, so shouldn't the sourcemap generation logic use those ranges when creating each attribute? 

Also, in `updateScript` you were returning a new SGScript without any of the original information previously...but with my clone method it should actually look more like the original script tag now.
![image](https://user-images.githubusercontent.com/2544493/105166471-7a40a680-5ae5-11eb-9072-570216864576.png)

Reply:
Hum oh right ok. I guess the question is whether a full clone is useful at all VS reusing objects, as it really happens during transcoding and the modified node isn't kept afterwards.
But it's up to you - it's fine.